src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,117 +1,227 @@
 
-  const componentClassName = clsx(
-    styles['checkbox'],
-    className,
-    inverted && styles['checkbox--inverted'],
+  return (
+    <svg
+      aria-hidden="true"
+      height={svgSizeRem}
+      viewBox={`0 0 ${svgViewBoxSize} ${svgViewBoxSize}`}
+      width={svgSizeRem}
+    >
+      <rect
+        className={styles['checkmark__outline']}
+        fill="none"
+        height="18"
+        rx="3"
+        stroke="currentColor"
+        strokeWidth="2"
+        width="18"
+        x="1"
+        y="1"
+      />
+      <path
+        className={styles['checkmark__content']}
+        d={checkmarkPath}
+        fill="none"
+      />
+    </svg>
   );
+};
+
+/**
+ * ```ts
+ * import {Checkbox} from "@chanzuckerberg/eds-components";
+ * ```
+ * ```ts
+ * import Checkbox, {CheckboxInput, Label} from '@chanzuckerberg/eds-components/lib/Checkbox';
+ * ```
+ *
+ * Checkbox control indicating if something is selected or unselected.
+ *
+ * Requires either a visible label or an accessible name.
+ */
+const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  (props, ref) => {
+    // All remaining props are passed to the `input` element
+    const { className, id, label, size, ...other } = props;
+
+    // When possible, use a visible label through the `label` prop instead.
+    // In rare cases where there's no visible label, you must provide an
+    // `aria-label` for screen readers.
+    if (
+      process.env.NODE_ENV !== 'production' &&
+      !label &&
+      !props['aria-label']
+    ) {
+      throw new Error('You must provide a visible label or aria-label');
+    }
+    const generatedId = useUID();
+    const checkboxId = id || generatedId;
+
+    return (
+      <div className={clsx(className, styles['checkbox'])}>
+        <CheckboxInput id={checkboxId} ref={ref} {...other} />
+        {label && <Label htmlFor={checkboxId} size={size} text={label} />}
+      </div>
+    );
+  },
+);
 
+Checkbox.displayName = 'Checkbox';
+
+/**
+ * Checkbox input element, exported for greater flexibility.
+ * You must provide an `id` prop and connect it to a visible label.
+ */
+const CheckboxInput = React.forwardRef<HTMLInputElement, InputProps>(
```

Comment:
so currently, their [guidance](https://github.com/chanzuckerberg/edu-design-system/blob/next/docs/CODE_GUIDELINES.md#exactly-one-component-per-directory-and-per-folder) says that sub-components should go in their own folder, e.g. `CheckboxInput/CheckboxInput.tsx`.

What do you think / is that do-able here? If it takes too much work to re-finagle we could file a ticket to come back to it later

Reply:
Hmmmmm, I'll take a quick stab at it

Reply:
Separated `Checkbox`, `CheckboxInput`, and `CheckboxLabel`. Lmk how you feel about it and the separated `CheckboxLabel`. Would be easy to revert since I committed separately